scapy/layers/inet.py

```diff
@@ -521,11 +521,28 @@ def hashret(self):
     def answers(self, other):
         if not isinstance(other, TCP):
             return 0
+        # RST packets don't get answers
+        if other.flags.R:
+            return 0
+        if self.flags.S:
+            # SYN packets without ACK are not answers
```

Comment:
Indeed four-way handshakes are valid with TCP:
SYN ->
        <- SYN
SYN/ACK ->
     <- ACK

However, I don't think that there is a need to support them.